### ISSUE TITLE
`no-duplicate-selectors` throws `postcss-selector-parser` error with SCSS nested property

### ISSUE THREAD

**Issue by JounQin** [stylelint/stylelint#6103]
### What steps are needed to reproduce the bug?

```scss
font: {
  size: 10px;
  weight: 400;
}
```

https://github.com/1stG/configs/tree/stylelint_scss

Branch `stylelint_scss`

`yarn && yarn lint:style`

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "@1stg/stylelint-config",
    "@1stg/stylelint-config/modules"
  ]
}
```

### How did you run Stylelint?

stylelint . --cache

### Which version of Stylelint are you using?

14.8.3

### What did you expect to happen?

No error

### What actually happened?

```log
Error: Expected a pseudo-class or pseudo-element.
    at /Users/JounQin/Workspaces/GitHub/configs/tests/test.scss:6:3
    at Root._error (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:174:16)
    at Root.error (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/selectors/root.js:43:19)
    at Parser.error (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:740:21)
    at Parser.expected (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:1133:19)
    at Parser.pseudo (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:875:19)
    at Parser.parse (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:1084:14)
    at Parser.loop (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:1043:12)
    at new Parser (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/parser.js:164:10)
    at Processor._root (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/processor.js:53:18)
    at Processor._runSync (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/processor.js:100:21)
    at Processor.processSync (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss-selector-parser/dist/processor.js:197:23)
    at normalize (/Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/rules/no-duplicate-selectors/index.js:164:26)
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/rules/no-duplicate-selectors/index.js:68:74
    at Array.map (<anonymous>)
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/rules/no-duplicate-selectors/index.js:68:56
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:96:18
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:55:18
    at Rule.each (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:41:16)
    at Rule.walk (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:52:17)
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:60:24
    at Root.each (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:41:16)
    at Root.walk (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:52:17)
    at Root.walkRules (/Users/JounQin/Workspaces/GitHub/configs/node_modules/postcss/lib/container.js:94:19)
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/rules/no-duplicate-selectors/index.js:54:8
    at /Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/lintPostcssResult.js:113:8
    at Array.map (<anonymous>)
    at lintPostcssResult (/Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/lintPostcssResult.js:104:18)
    at lintSource (/Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/lintSource.js:91:8)
    at async /Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/standalone.js:228:27
    at async Promise.all (index 12)
    at async standalone (/Users/JounQin/Workspaces/GitHub/configs/node_modules/stylelint/lib/standalone.js:267:22)
```

### Does the bug relate to non-standard syntax?

Yes, but I've already used `customSyntax: 'postcss-scss'`, and it does not throw on `14.8.2`

### Proposal to fix the bug

<img width="1037" alt="image" src="https://user-images.githubusercontent.com/8336744/169698705-3156a3aa-c9b4-403c-8fa3-b385de9f20da.png">

It could be related to `normalize-selector` been removed.

**Comment by novogrammer** [comment]
Same here.

failed when stylelint@14.8.3 and stylelint-config-twbs-bootstrap@3.0.1

stylelint@14.8.2 is ok

https://github.com/novogrammer/test-stylelint-20220522

**Comment by ybiquitous** [comment]
@JounQin @novogrammer Thanks for reporting the issue with the reproduction and using the template.

I can confirm the regression on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjJYA1gOYBOEArplPGipADYSm0DEAZgAycoC+A3Fl5YQAGhABnAJ6YALkgAeCSWAkSx4CJnYBLYsozYU6EKQoMYEk7UM4cJzBAC0UCgAcGOsElkwnEmAswWWYrBBRZMxgsHGFMXhBeIA).

This regression with the `no-duplicate-selectors` rule seems related to the SCSS nested syntax like `a { background: { color: red } }`.

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ybiquitous** [comment]
Related to #6068

**Comment by ybiquitous** [comment]
The cause seems that `postcss-selector-parser` fails to parse a selector ending with `:` like `background:`. See the code below:

```js
// a.js
postcss = require('postcss');
selectorParser = require('postcss-selector-parser');

code = 'a { background: {} }';

postcss.parse(code).walkRules((rule) => console.log(`selector: "${rule.selector}"`));

console.log('parsed: ' + selectorParser().processSync('a', { lossless: false }));
console.log('parsed: ' + selectorParser().processSync('background:', { lossless: false }));
```

Run:

```console
$ node a.js
selector: "a"
selector: "background:"
parsed: a
/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:740
    throw this.root.error(message, opts);
    ^

Error: Expected a pseudo-class or pseudo-element.
    at Root._error (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:174:16)
    at Root.error (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/selectors/root.js:43:19)
    at Parser.error (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:740:21)
    at Parser.expected (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:1133:19)
    at Parser.pseudo (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:875:19)
    at Parser.parse (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:1084:14)
    at Parser.loop (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:1043:12)
    at new Parser (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/parser.js:164:10)
    at Processor._root (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/processor.js:53:18)
    at Processor._runSync (/Users/koba/git/stylelint/stylelint/node_modules/postcss-selector-parser/dist/processor.js:100:21)

Node.js v18.2.0
```

To avoid such a parse error, I think we should use `isStandardSyntaxSelector()` for a non-standard selector syntax:

https://github.com/stylelint/stylelint/blob/2b4147b8ec5e1aa1136ed858004b53635c022ec3/lib/utils/isStandardSyntaxSelector.js#L22-L25

**Comment by JounQin** [comment]
@ybiquitous Please help to review #6106
